packages/react-router/src/utils.ts

```diff
@@ -205,7 +205,7 @@ function hasObjectPrototype(o: any) {
   return Object.prototype.toString.call(o) === '[object Object]'
 }
 
-export function isPlainArray(value: unknown) {
+export function isPlainArray(value: unknown): value is Array<unknown> {
```

Comment:
do we even need this with TS5.5 [Inferred Type Predicates](https://devblogs.microsoft.com/typescript/announcing-typescript-5-5/#inferred-type-predicates)?

=> worth trying to upgrade to TS5.5

Reply:
We can't use TS 5.5 since we've currently stated that the minimum [requirement](https://tanstack.com/router/latest/docs/framework/react/installation#requirements) for the TS version is 5.0 on the website.

We can change to TS 5.5 ofc, but I believe this is something that'd normally be a part of a major release.